app/services/publish_vacancy.rb

```diff
@@ -1,15 +1,19 @@
 class PublishVacancy
-  def initialize(vacancy, current_user)
-    @vacancy = vacancy
+  attr_accessor :current_organisation, :current_user, :vacancy
+
+  def initialize(vacancy, current_user, current_organisation)
+    @current_organisation = current_organisation
     @current_user = current_user
+    @vacancy = vacancy
   end
 
   def call
-    return false unless @vacancy.valid?
+    return false unless vacancy.valid?
 
-    @vacancy.publisher_user_id = @current_user.id
-    @vacancy.status = :published
-    @vacancy.state = 'edit_published'
-    @vacancy.save
+    vacancy.publisher_organisation_id = current_organisation.id
```

Comment:
You should be able to do:
`vacancy.publisher_organisation = current_organisation`
same as for next line